### PR TITLE
Add (POST) products

### PR DESCRIPTION
**Pull request by OHubert90**

### CODE REVIEW

**Comment by josch87** [comment]
Hi. Unfortunately, this pull request is not linked to an issue (e.g. through the keyword closes or fixes) and does not have a description. For which problem or task are you offering a solution here?

File: backend/src/main/java/de/neuefische/backend/config/CorsConfig.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package de.neuefische.backend.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class CorsConfig {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/config/CorsConfig.java]: Remove
add frontend vite proxy

Comment (Flooooooooooorian) [reply backend/src/main/java/de/neuefische/backend/config/CorsConfig.java]: ` server: {
    proxy: {
      '/api': {
        target: 'http://localhost:8080',
      }
    }
  }`

File: frontend/src/productsData.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface Product {
+    id: string;
+    name: string;
+    stock: number;
+    price: number;
+}
+
+export const fetchProducts = async (): Promise<Product[]> => {
+    const response = await axios.get('http://localhost:8080/api/products');
```
Comment (Flooooooooooorian) [inline frontend/src/productsData.ts]: Use relative url

File: backend/src/main/java/de/neuefische/backend/controller/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @PostMapping
+    public Products addProduct(@RequestBody Products product) {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/controller/ProductController.java]: Add DTO

File: frontend/src/AddProductForm.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import './AddProductForm.css';
+
+interface Product {
+    id?: string;
```
Comment (Flooooooooooorian) [inline frontend/src/AddProductForm.tsx]: add dto

File: frontend/src/AddProductForm.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import './AddProductForm.css';
+
+interface Product {
```
Comment (Flooooooooooorian) [inline frontend/src/AddProductForm.tsx]: extract to seperate file to use exports

File: frontend/src/AddProductForm.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    const [products, setProducts] = useState<Product[]>([]);
+    const [errorMessage, setErrorMessage] = useState<string>('');
+
+    useEffect(() => {
+        fetchProducts();
+    }, []);
+
+    const fetchProducts = async () => {
+        try {
+            const response = await axios.get('http://localhost:8080/api/products');
```
Comment (Flooooooooooorian) [inline frontend/src/AddProductForm.tsx]: relative url

File: frontend/src/AddProductForm.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        setProduct({
+            ...product,
+            [name]: value,
+        });
+    };
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        try {
+            await axios.post('http://localhost:8080/api/products', product);
```
Comment (Flooooooooooorian) [inline frontend/src/AddProductForm.tsx]: relative url

File: frontend/src/AddProductForm.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+            fetchProducts();
```
Comment (Flooooooooooorian) [inline frontend/src/AddProductForm.tsx]: use return from post

File: frontend/src/AddProductForm.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+            alert('Product added successfully');
+            setProduct({ name: '', stock: 0, price: 0.0 }); // Reset form
```
Comment (Flooooooooooorian) [inline frontend/src/AddProductForm.tsx]: remove comment

File: frontend/src/productsData.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import axios from 'axios';
+
+export interface Product {
+    id: string;
+    name: string;
+    stock: number;
+    price: number;
+}
+
+export const fetchProducts = async (): Promise<Product[]> => {
```
Comment (Flooooooooooorian) [inline frontend/src/productsData.ts:16]: duplicate

**Review by Flooooooooooorian — COMMENTED** [review]
Keep App.tsx

**Pull request merged**